dcql/src/dcql-parser/dcql-meta-query-result.ts

```diff
@@ -70,27 +70,39 @@ const getW3cVcMetaParser = (credentialQuery: DcqlCredentialQuery.W3cVc) => {
   })
 }
 
-export const getMetaParser = (credentialQuery: DcqlCredentialQuery) => {
+// NOTE: is it a bit of "cheating" to use the credential as input here. We need it
+// to be able to discern between legacy SD-JWT VCs and W3C VCs since they use the
+// same format 'vc+sd-jwt'. When removing legacy SD-JWT VCs, we can also remove the
+// credential input here.
+export const getMetaParser = (credentialQuery: DcqlCredentialQuery, credential?: DcqlCredential) => {
   if (credentialQuery.format === 'mso_mdoc') {
     return getMdocMetaParser(credentialQuery)
   }
 
-  if (credentialQuery.format === 'dc+sd-jwt' || credentialQuery.format === 'vc+sd-jwt') {
+  if (credentialQuery.format === 'dc+sd-jwt') {
     return getSdJwtVcMetaParser(credentialQuery)
   }
 
+  if (credentialQuery.format === 'vc+sd-jwt') {
+    if (credential && 'vct' in credential) {
+      return getSdJwtVcMetaParser(credentialQuery as DcqlCredentialQuery.SdJwtVc)
+    }
+
+    return getW3cVcMetaParser(credentialQuery as DcqlCredentialQuery.W3cVc)
+  }
```

Comment:
I think the credential is not neccesary to be passed.

Since meta is required on w3c creds, if type_values is in the query meta we know to use the w3c vc parser.

Reply:
You're right. I had looked at the meta first, but I was under the impression that it was actually optional. But indeed, it is mandatory, so it's fine to use that.